libs/datamodel/connectors/datamodel-connector/src/error.rs

```diff
@@ -1,2 +1,77 @@
-#[allow(unused)]
-enum ConnectorError {}
+use crate::scalars::ScalarType;
+use std::fmt;
+use thiserror::Error;
+
+#[derive(Debug, Error, Clone, PartialEq)]
+pub struct ConnectorError {
```

Comment:
This is a good start. You should probably evolve this to a richet structure similar to this one: https://github.com/prisma/prisma-engines/blob/master/migration-engine/connectors/migration-connector/src/error.rs#L8

Right now the error details are an opaque string and a client would not be able to react to specific errors through pattern matching.